File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/F08.P2.kt
fun guarded(a: Int, b: Int): Int {
    // BEGIN PROBE F08.P2
    val q = if (b == 0) 0 else a / b
    return q
    // END PROBE F08.P2
}

fun main() {
    println(guarded(7, 0))
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F08.P2.rs
fn guarded(a: i32, b: i32) -> i32 {
// BEGIN PROBE F08.P2
let q = a.checked_div(b).unwrap_or(0);
q
// END PROBE F08.P2
}

fn main() {
    println!("{}", guarded(7, 0));
}
